src/ddscxx/src/org/eclipse/cyclonedds/core/cond/WaitSetDelegate.cpp

```diff
@@ -66,7 +73,9 @@ org::eclipse::cyclonedds::core::cond::WaitSetDelegate::wait(
     const dds::core::Duration& timeout)
 {
     dds_duration_t c_timeout = org::eclipse::cyclonedds::core::convertDuration(timeout);
+    org::eclipse::cyclonedds::core::ScopedObjectLock scopedLock(*this);
     const size_t sz = conditions_.size();
```

Comment:
I assume this size should be fine even if some conditions are removed later, since its upper bound size for the conditions. Is this assumption correct?

Reply:
You could also add one more while waiting, but too it is not a problem: it simply means that the call to `wait` can return that more conditions triggered then it could return in the `attach` array. That's supported in the C code, and the code below allows for it as well (line 91 in the new line numbering).